### PR TITLE
#23 join class functionality

### PR DESCRIPTION
**Pull request by BredeFK**

### CODE REVIEW

**Comment by BredeFK** [comment]
[#23 - Join Class Functionality](https://trello.com/c/lGOGylxO/23-23-join-class-functionality)

File: config/database.sql [hunk 1]
```diff
@@ -57,7 +57,7 @@ CREATE TABLE `assignments`
 
 CREATE TABLE `course`
 (
-  `id`          int(11)                                        NOT NULL,
+  `id`          varchar(64)                                    NOT NULL,
```
Comment (kongebra) [inline config/database.sql]: Primary key should be an int, for indexing, and the new "hash" should be called something else

File: config/database.sql [hunk 2]
```diff
@@ -80,7 +80,7 @@ CREATE TABLE `logs`
   `timestamp`    datetime                           NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `activity`     varchar(32) COLLATE utf8_danish_ci NOT NULL,
   `assignmentid` int(11)                                     DEFAULT NULL,
-  `courseid`     int(11)                                     DEFAULT NULL,
+  `courseid`     varchar(64)                                     DEFAULT NULL,
```
Comment (kongebra) [inline config/database.sql]: Should still be an int(11)

File: config/database.sql [hunk 3]
```diff
@@ -150,8 +150,8 @@ CREATE TABLE `submissions`
 
 CREATE TABLE `usercourse`
 (
-  `userid`   int(11) NOT NULL,
-  `courseid` int(11) NOT NULL
+  `userid`   int(11)     NOT NULL,
+  `courseid` varchar(64) NOT NULL
```
Comment (kongebra) [inline config/database.sql]: Should still be an int(11)

File: config/database.sql [hunk 4]
```diff
@@ -36,12 +36,12 @@ SET time_zone = "+01:00";
...
 CREATE TABLE `assignments`
 (
-  `id`       int(11)    NOT NULL,
-  `courseid` int(11)    NOT NULL,
-  `created`  timestamp  NOT NULL                DEFAULT CURRENT_TIMESTAMP,
-  `due`      date       NOT NULL,
-  `peer`     tinyint(1) NOT NULL                DEFAULT '0',
-  `auto`     tinyint(1) NOT NULL                DEFAULT '0',
+  `id`       int(11)     NOT NULL,
+  `courseid` varchar(64) NOT NULL,
```
Comment (kongebra) [inline config/database.sql]: Should still be an int(11)

File: config/database.sql [hunk 5]
```diff
@@ -294,7 +295,7 @@ ALTER TABLE `assignments`
 -- AUTO_INCREMENT for table `course`
 --
 ALTER TABLE `course`
-  MODIFY `id` int(11) NOT NULL AUTO_INCREMENT;
+  MODIFY `id` varchar(64) NOT NULL;
```
Comment (kongebra) [inline config/database.sql]: Should be an int(11), and primary key.

File: config/database.sql (hunk 1 above)
Comment (kongebra) [reply config/database.sql]: This new unique generated id, should be a new column

**Pull request merged**